src/Voron/Impl/PagerState.cs

```diff
@@ -130,20 +202,35 @@ private void ReleaseInternal()
                 File.Dispose();
                 File = null;
             }
-
-            _released = true;
         }
 
 #if DEBUG_PAGER_STATE
         public ConcurrentQueue<string> AddedRefs = new ConcurrentQueue<string>();
 #endif
         public void AddRef()
         {
-            if (_released)
-                ThrowInvalidPagerState();
+            int currentRefCount;
+            int newRefCount;
+
+            // Use atomic compare-and-swap to safely increment the reference count,
+            // ensuring consistency even with concurrent modifications by other threads.
+            do
+            {
+                currentRefCount = _refs;
+
+                // Prevent adding a reference to an invalid or already disposed state,
+                // protecting against undefined behavior and ensuring references are only added to valid, active objects.
+                if (currentRefCount < 0)
+                    ThrowInvalidPagerState();
```

Comment:
Here as well?

Reply:
No because here it can be 0 (when just initialized).

Reply:
However, that question actually uncovered a race condition :) ... bug fix incoming.